vizro-ai/src/vizro_ai/dashboard/nodes/core_builder/plan.py

```diff
@@ -0,0 +1,321 @@
+        filter_prompt = (
+            f"Create a filter from the following instructions: {self.control_description}. Do not make up "
+            f"things that are optional and DO NOT configure actions, action triggers or action chains."
+            f" If no options are specified, leave them out."
+        )
+        try:
+            _df_schema, _df_sample = (
+                df_metadata[self.data_frame]["df_schema"],
+                df_metadata[self.data_frame]["df_sample"],
+            )
+            _df_cols = list(_df_schema.keys())
+        # when wrong dataframe name is given
+        except KeyError:
+            logger.info(f"Dataframe {self.data_frame} not found in metadata, returning default values.")
+            return None
+
+        try:
+            result_proxy = create_filter_proxy(
+                df_cols=_df_cols, df_sample=_df_sample, available_components=available_components
+            )
+            proxy = _get_model(query=filter_prompt, model=model, result_model=result_proxy, df_metadata=df_metadata)
+            logger.info(
+                f"`Control` proxy: {proxy.dict()}"
+            )  # when wrong column name is given, `AttributeError: 'ValidationError' object has no attribute 'dict'``
+            actual = Filter.parse_obj(
+                proxy.dict(
+                    exclude={"selector": {"id": True, "actions": True, "_add_key": True}, "id": True, "type": True}
+                )
+            )
+            # del model_manager._ModelManager__models[proxy.id]  # TODO: This is very wrong and needs to change
+
+        except ValidationError as e:
+            logger.info(f"Build failed for `Control`, returning default values. Error details: {e}")
+            return None
+
+        return actual
+
+
+class LayoutProxyModel(BaseModel):
+    """Proxy model for Layout."""
+
+    grid: List[List[int]] = Field(..., description="Grid specification to arrange components on screen.")
+
+    @validator("grid")
+    def validate_grid(cls, grid):
+        """Validate the grid."""
+        if len({len(row) for row in grid}) > 1:
+            raise ValueError("All rows must be of same length.")
+
+        # Validate grid type and values
+        unique_grid_idx = _get_unique_grid_component_ids(grid)
+        if 0 not in unique_grid_idx or not np.array_equal(unique_grid_idx, np.arange((unique_grid_idx.max() + 1))):
+            raise ValueError("Grid must contain consecutive integers starting from 0.")
+
+        # Validates grid areas spanned by components and spaces
+        component_grid_lines, space_grid_lines = _get_grid_lines(grid)
+        _validate_grid_areas(component_grid_lines + space_grid_lines)
+        return grid
+
+
+class Layout(BaseModel):
+    """Layout plan model."""
+
+    layout_description: str = Field(
+        ...,
+        description="Description of the layout. Include everything that seems to relate"
+        " to this layout AS IS. If layout not provided, specify `NO layout`.",
```

Comment:
Could we handle this as part of an optional field? So that layout is naturally included or not